optlib/cmake.ctags

```diff
@@ -132,19 +149,33 @@
 --_mtable-extend-CMake=variable+skipToName
 --_mtable-regex-CMake=variable/([A-Za-z0-9_.-]+)(#)/\1/v/{tleave}{_advanceTo=2start}
 
+# when a variable is defined inside a function, we reference its scope unless it
+# has a PARENT_SCOPE argument; unfortunately the regex for this can backtrack a
+# lot and the current ctags regex engine doesn't do lazy or atomic/possessive
+# captures to avoid it, but in practice this should be ok
+--_tabledef-CMake=inVariable
+--_mtable-regex-CMake=inVariable/[^")]+(("(\\"|[^"])*")([^")]+("(\\"|[^"])*"))*)[ \t\n]PARENT_SCOPE[# \t\n)]//{tjump=variable}{_advanceTo=0start}
+--_mtable-regex-CMake=inVariable/([A-Za-z0-9_.-]+)[ \t\n\)]+/\1/v/{tleave}{scope=ref}
+--_mtable-regex-CMake=inVariable/([A-Za-z0-9_.-]+)(#)/\1/v/{tleave}{scope=ref}{_advanceTo=2start}
+
+--_mtable-regex-CMake=variableScoped/[A-Za-z0-9_.-]+[# \t\n\)]//{tjump=inVariable}{_advanceTo=0start}
+--_mtable-extend-CMake=variableScoped+skipToName
+
+
 #
 # function
 #
---_mtable-regex-CMake=function/([A-Za-z_][A-Za-z0-9_]*)[ \t\n\)]+/\1/f/{tleave}
+--_mtable-regex-CMake=function/([A-Za-z_][A-Za-z0-9_]*)([# \t\n\)])/\1/f/{_advanceTo=2start}{tjump=inFunction}{scope=push}
 --_mtable-extend-CMake=function+skipToName
---_mtable-regex-CMake=function/([A-Za-z_][A-Za-z0-9_]*)(#)/\1/f/{tleave}{_advanceTo=2start}
 
 #
-# macro
+# macro: unlike functions, technically macros don't create scopes for any
+# variables set in them - the scope is whatever it is when the macro is invoked.
+# So we're not going to push a scope nor go into the 'inFunction' table.
 #
```

Comment:
MY TODO:
Though a macro doesn't make a scope, filling end: field for it may be useful.
A new scope action something like `{scope=pushOnlyPosition}` is needed.
It requires changes in the main part.